### ISSUE TITLE
Multi-line search

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#13801]
### Problem

Only single-line search is currently possible. Selecting multi-line text does not populate the search box as it multi-line searach is not supported.

### Proposed Solution

Implement multi-line search.

### Additional context

This is pretty standard feature in code editors.

**Comment by krassowski** [comment]
Three code fragments need to be changed:

1. We need to add multi-line flag here:

https://github.com/jupyterlab/jupyterlab/blob/31886fdee1092646a782feb947877937e30c2170/packages/documentsearch/src/searchmodel.ts#L318-L344

2. We need to remove the logic which prevents select-on-search for multi-line selections

https://github.com/jupyterlab/jupyterlab/blob/31886fdee1092646a782feb947877937e30c2170/packages/fileeditor/src/searchprovider.ts#L70-L78

https://github.com/jupyterlab/jupyterlab/blob/31886fdee1092646a782feb947877937e30c2170/packages/notebook/src/searchprovider.ts#L186-L195

3. We need to replace the `input`s for query and replace text with `textarea`s:

https://github.com/jupyterlab/jupyterlab/blob/31886fdee1092646a782feb947877937e30c2170/packages/documentsearch/src/searchview.tsx#L89-L98

https://github.com/jupyterlab/jupyterlab/blob/31886fdee1092646a782feb947877937e30c2170/packages/documentsearch/src/searchview.tsx#L143-L151

**Comment by c00kie123** [comment]
Hey I can try working on this

**Comment by krassowski** [comment]
Feel welcome to work on it. Please open a (draft) PR when you have something ready. Please see our contributing guidelines: https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html

**Comment by c00kie123** [comment]
> Additionally we would want to handle enter vs shift/ctrl enter. Enter should search, shift enter add new line.

For this part, I am thinking we can do this.
```
onKeyDown={e => {
    if (e.key === 'Enter' && e.shiftKey) {
        // handle shift+enter (add new line)
        e.target.value += '\n'
    } else if (e.key === 'Enter') {
        // handle enter (search)
    }
}}
```
What do you think of the shift+enter function and what we can add to handle enter(search)? Let me know thanks!

**Comment by c00kie123** [comment]
Ignore PR#13809. I forgot that I had to draft it first

**Comment by andrii-i** [comment]
Hey @c00kie123! You can put any open PR into a draft status and back at any time. Here's a [link](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#converting-a-pull-request-to-a-draft) to github docs on how. Didn't know it myself at first so wanted to let you know just in case

**Comment by krassowski** [comment]
Fixed by #13810.
